tsfile/src/main/java/org/apache/iotdb/tsfile/read/filter/operator/Like.java

```diff
@@ -51,7 +51,28 @@ public Like(String value, FilterType filterType) {
     this.value = value;
     this.filterType = filterType;
     try {
-      this.pattern = Pattern.compile(this.value);
+      String regexpStr = ".^$*+?{}[]|()";
+      StringBuilder patternStrBuild = new StringBuilder();
+      patternStrBuild.append("^");
```

Comment:
Add some comments about where the main idea of this part comes from? :>
https://codereview.stackexchange.com/questions/36861/convert-sql-like-to-regex/36864